### ISSUE TITLE
Import does nothing after selecting file

### ISSUE THREAD

**Issue by Zerotwistknife7** [grokability/snipe-it#12926]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

When I try to upload a csv file on the import page, nothing happens. No loading bar, absolutely nothing appears to happen. I also tried uploading the Assets-sample.csv from the Snipe-IT documentation.

I enabled debug mode in .env and I can't seem to find anything in the debug menus on the bottom of the page. None of the tab numbers change when trying to upload a file.

I also checked Chrome developer console and nothing appears to happen there either.

### Reproduction steps

- I go to the snipe-it-url/import page.
- I click on Select File... and windows file explorer window opens.
- I select a simple csv file name assets.csv (confirmed no special characters or anything strange in file).
- Click on Open, file explorer window closes.
- Nothing appears to happen on the Import page.

### Expected behavior

When testing on another Snipe-IT instance that is installed on Docker, I see a loading bar and the file then appears in the list below.

### Screenshots

![image](https://user-images.githubusercontent.com/17997610/234677361-19e62795-7c6b-4cba-beea-978212342fee.png)

### Snipe-IT Version

v6.0.14 build 9715 (g8b70a7f21)

### Operating System

Ubuntu Server 22.04.2

### Web Server

Apache2 I think?

### PHP Version

8.1.2-1ubuntu2.11

### Operating System

Windows 10 Pro

### Browser

Google Chrome

### Version

Chrome is up to date Version 112.0.5615.138 (Official Build) (64-bit)

### Error messages

```shell
Only errors I'm seeing are

[2023-04-24 19:19:02] production.ERROR: Your APP_URL in your .env is misconfigured - it is: . Many things will work strangely unless you fix it. 

Note: When I set APP_URL, it breaks my images by adding APP_URL into the image path. for example: http://assets.domain.com/assets.domain.com/uploads/models/assetmodel-image-1-yrwBtpsSpF.png
If I leave APP_URL=null, everything works fine.
```

### Additional context

This is a fresh installation, I followed the directions I found here: [https://syncbricks.com/snipe-it-instsallation-ubuntu-20-04/](https://syncbricks.com/snipe-it-instsallation-ubuntu-20-04/)

I'm hoping the import issue is just a simple permission that I'm missing somewhere. I can upload images and backups just fine, only thing that doesn't seem to work is uploading Imports.

**Comment by snipe** [comment]
What are you setting your APPL_URL to, and are you running behind a proxy of any kind?

**Comment by Zerotwistknife7** [comment]
> What are you setting your APPL_URL to, and are you running behind a proxy of any kind?

APP_URL=null
I tested with it set to my assets.domain.com and uploading import still didn't seem to do anything, so I set it back to null for now.

I tried uploading via http://IP/import and http://assets.[domain].com/import and still nothing happens when selecting a file for importing.

I am not running behind any proxy.

**Comment by snipe** [comment]
APP_URL being null is definitely going to cause you problems - that's why we explicitly tell you not to do it. And you checked your application log as well, right?

**Comment by Zerotwistknife7** [comment]
> APP_URL being null is definitely going to cause you problems - that's why we explicitly tell you not to do it. And you checked your application log as well, right?

Sorry for my ignorance, Where would I find the application log?

**Comment by snipe** [comment]
https://snipe-it.readme.io/docs/getting-help

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/getting-help">Getting Help</a></strong></div><div>If you're having trouble setting up Snipe-IT, don't worry!</div></blockquote>

**Comment by Zerotwistknife7** [comment]
Yeah still only seeing the APP_URL errors all the way up, nothing else is shown.

`[2023-04-26 16:09:05] production.ERROR: Your APP_URL in your .env is misconfigured - it is: . Many things will work strangely unless you fix it.`

EDIT: Okay so I figured out why my APP_URL was misconfigured. I guess I forgot the http:// in from of assets.domain.com

I get no more APP_URL errors, but still can not upload an import file. I just downloaded the sample-assets.csv and tried importing it. Again nothing happens after selecting the file.

**Comment by marcusmoore** [comment]
Are there new messages in the application log or the browser's console?

**Comment by Zerotwistknife7** [comment]
No new errors since fixing the APP_URL. I just tried uploading the sample-assests.csv in an Microsoft Edge incognito window and see absolutely nothing happen after selecting the file.

![VX8qpEgZip](https://user-images.githubusercontent.com/17997610/234984837-8c1e0078-5eb8-4866-aa37-e2664ebf0f33.png)

![eYsW7x8X1v](https://user-images.githubusercontent.com/17997610/234984846-cc794cdf-0fe6-4aa7-9ada-92d8024f81e3.png)

Everything else seems to work except the import page and I have a pretty large list of assets waiting on getting this working. At this point I'm getting ready to go into ESXi and just delete this Ubuntu VM and start over.

**Comment by marcusmoore** [comment]
> At this point I'm getting ready to go into ESXi and just delete this Ubuntu VM and start over.

I understand your frustration. This can be an annoying process but let's see if we can get you up and running 😄 

I noticed something in your screenshot:
<img width="549" alt="console errors" src="https://user-images.githubusercontent.com/1141514/234989734-3648bcf6-7560-4267-a2d4-856c2a804e15.png">

I see you're on `v6.0.14 build 9715`. Can you update to the latest version and see if that helps? I'm pretty sure I saw that console error message at some point during the 6.0.x branch.

**Comment by Zerotwistknife7** [comment]
> I see you're on `v6.0.14 build 9715`. Can you update to the latest version and see if that helps? I'm pretty sure I saw that console error message at some point during the 6.0.x branch.

I realize I should have moved the window over a little or made it smaller.
![image](https://user-images.githubusercontent.com/17997610/235006066-fd201d53-eed5-41c7-be0e-3f93d3641bec.png)

Forgive my ignorance, (I'm a Windows sysadmin) How would I go about updating my "var/www/Snipe-IT" installation?

_I installed using these directions. [https://syncbricks.com/snipe-it-instsallation-ubuntu-20-04/](https://syncbricks.com/snipe-it-instsallation-ubuntu-20-04/)_

I'm pretty sure I don't want to do a git clone of the entire github repo, what should I type to update my installation instead?

Thank for all your help so far.

<blockquote><div><img src="https://syncbricks.com/wp-content/uploads/2020/05/cropped-syncbricks-logo-512x512-1-192x192.png" height="14"> SYNCBRICKS -  Information Technology for everyone</div><div><strong><a href="https://syncbricks.com/snipe-it-instsallation-ubuntu-20-04/">Snipe IT Instsallation Ubuntu 20.04 » SYNCBRICKS - Information Technology for everyone</a></strong></div><div>Snipe-IT is one of the open-source software for IT Assets Management that gives Transparency, security, and oversight to your IT assets. Installation can be nightmare but I will explain you the installation in few steps. Contents 1. Install Ubuntu Server 20.04 You can do this either on a physical server or VM Server. Open SSH ... <a title="Snipe IT Instsallation Ubuntu 20.04" class="read-more" href="https://syncbricks.com/snipe-it-instsallation-ubuntu-20-04/" aria-label="More on Snipe IT Instsallation Ubuntu 20.04">Read more</a></div></blockquote>

**Comment by marcusmoore** [comment]
No worries! Our [install instructions](https://snipe-it.readme.io/docs/downloading) recommend installing via git and it looks like the guide you linked says to do a `git clone` as well.

If you followed that process and installed via git then you should be able to run `php upgrade.php` to update to the latest. **Please see the details in the [upgrade guide](https://snipe-it.readme.io/docs/upgrading) before running that command.**

If git is not available then check out the "Upgrading Manually" section in the upgrade guide for that process.

**Comment by Zerotwistknife7** [comment]
> you should be able to run `php upgrade.php` to update to the latest.

Here we go, now we're getting somewhere. I see folder permission denied errors in Steps 3-6 of the update. [php upgrade log.txt](https://github.com/snipe/snipe-it/files/11349207/php.upgrade.log.txt)

I tried fixing the directory permissions with directions I found here, [https://snipe-it.readme.io/docs/configuration#required-set-directory-permissions](https://snipe-it.readme.io/docs/configuration#required-set-directory-permissions)

It told me the apache group didn't exist so I created it and continued with the commands.

`sudo groupadd apache`
`sudo useradd -g apache snipeit`
`sudo chown -R snipeit storage public/uploads bootstrap/cache`
`sudo chmod -R 775 storage`
`sudo chmod -R 775 public/uploads`
`sudo chmod -R 775 bootstrap/cache`

It gave me a `chmod: changing permissions ...blah... Operation not permitted`, but it looks like I was able to set the permissions using `sudo`.

I tried to upgrade again, but I still get the same permissions denied messages as before.

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/configuration#required-set-directory-permissions">Configuration</a></strong></div><div>This section is where you edit the Snipe-IT configuration file to reflect your own settings, such as your database credentials, mail server, preferred language, timezone, and so on. Some of the settings are optional, some are required.Don't be intimidated by the length of this page. The configuratio...</div></blockquote>

**Comment by marcusmoore** [comment]
Is the snipeit user the same user you are running the upgrade script as?

**Comment by Zerotwistknife7** [comment]
> Is the snipeit user the same user you are running the upgrade script as?

Holy crap, that's it! I never correlated 1. setting the permissions above for the snipeit user and 2. needing to actually run it as the snipeit user.

Come to think of it, I've never even logged in as $user snipeit. I had to reset the passwd just in order to login as snipeit.

First time running php upgrade.php as snipeit, Step 3 said I still didn't have write permissions to the folders. **Fixed.**

Second time I noticed an error stating that I had no /home/ directory. **Fixed.**

Third time detected `dubious ownership in repository` at least it gives a command to run. `git config --global --add safe.directory /var/www/snipe-it` **Fixed.**

Final time I ran upgrade, it said I didn't have permission to run git. `error: cannot open .git/FETCH_HEAD: Permission denied` but was able to resolve that with `sudo chown -R snipeit .git/` **Fixed.**

What an adventure and "fun" way to learn linux, haha.

![image](https://user-images.githubusercontent.com/17997610/235578254-8d9498c7-766c-43c3-9d37-a829818b3228.png)

Thank you @snipe @marcusmoore for all your help and making a great software.
